source/common/json/json_loader.h

```diff
@@ -1,12 +1,30 @@
 #pragma once
 
 #include "envoy/common/exception.h"
+#include "envoy/common/pure.h"
 
 #include "common/common/non_copyable.h"
 
-struct json_t;
-
 namespace Json {
+class Object;
```

Comment:
split object out into public include and only define the factory stuff in this file.